presto-parser/src/main/java/com/facebook/presto/sql/tree/Identifier.java

```diff
@@ -87,12 +87,12 @@ public boolean equals(Object o)
         }
 
         Identifier that = (Identifier) o;
-        return Objects.equals(value, that.value);
```

Comment:
@findepi To me it is like comparing `t.a` with `a` which comes from relation `t`. Simply comparing expression will return `false`, and Presto currently is unable to recognize that these two expressions are equivalent in places you refer.

I think we need to handle the case expression equivalence differently (like introducing expression equivalence comparator with usage of `Scope` and `NodeRef` as we were talking about offline) and leave expressions as they are (leave equals/hashCode to compare all the fields).  We still need to be able  to distinguish `"AlA"` from `"aLa"`  from `ala`.

What do yo think?